backend/src/whatsapp/middlewares/whatsapp.middleware.ts

```diff
@@ -0,0 +1,53 @@
+import { loggerWithLabel } from '@core/logger'
+import { Handler, Request, Response } from 'express'
+import { CredentialService } from '@core/services'
+import { WhatsappService } from '@whatsapp/services'
+
+export interface WhatsappMiddleware {
+  getCredentialsFromBody: Handler
+  getCampaignDetails: Handler
+
+  sendMessage: Handler
+}
+
+export const InitWhatsappMiddleware = (
+  credentialService: CredentialService
+): WhatsappMiddleware => {
+  const logger = loggerWithLabel(module)
+  const getCredentialsFromBody = async (
+    req: Request,
+    res: Response
+  ): Promise<void | Response> => {
+    logger.info(req, res, credentialService)
+  }
+
+  const getCampaignDetails = async (
+    req: Request,
+    res: Response
+  ): Promise<void | Response> => {
+    const { campaignId } = req.params
+    return res.json({ campaignId })
+  }
+
+  const sendMessage = async (
+    req: Request,
+    res: Response
+  ): Promise<void | Response> => {
+    const { recipient, from, content } = req.body
+    try {
+      const resp = await WhatsappService.sendMessage(from, recipient, content)
+      return res.json(resp)
+    } catch (e) {
+      logger.error({
+        message: 'Something went wrong with sending a message',
+        error: e,
+      })
+      return res.status(401).json({ message: e })
```

Comment:
Shouldn't bee 401, should be a 500 if WA is down or something (catastrophic), or codes depending the specific errors thrown from WA